tests/test_log.c

```diff
@@ -107,7 +120,7 @@ void TestLog_append_is_not_empty(CuTest * tc)
     raft_cbs_t funcs = {
         .log_offer = __log_offer
     };
-    log_set_callbacks(l, &funcs, r);
+    log_set_callbacks(l, &funcs, tc);
```

Comment:
This fixes an upstream bug that uses r as tc. See the cast of r to tc inside __log_offer. If I understand correctly, that is very bad.

In other callback implementations in this file, I take a shortcut by using the assert.h facilities instead of those from CuTest, because of the lack of tc in the callback implementations. Debugging regressions will be a bit tedious though, for one will have to add print statements or employ a debugger.